--- CMakeLists.txt	2014-04-10 19:48:36.711851129 +0900
+++ CMakeLists.txt	2014-04-10 19:49:09.860418220 +0900
@@ -71,6 +71,13 @@
 include_directories(${DB8_INCLUDE_DIRS})
 webos_add_compiler_flags(ALL ${DB8_CFLAGS_OTHER})
 
+if (NOT DEFINED USE_PMLOG OR USE_PMLOG)
+    # -- check for pmLogLib
+    pkg_check_modules(PMLOG REQUIRED PmLogLib>=3)
+    include_directories(${PMLOG_INCLUDE_DIRS})
+    webos_add_compiler_flags(ALL ${PMLOG_CFLAGS_OTHER} -DUSE_PMLOG)
+endif()
+
 find_package(Boost REQUIRED COMPONENTS filesystem system)
 include_directories(${Boost_INCLUDE_DIRS})
 
@@ -88,6 +95,7 @@
 			${LS2_LDFLAGS}
 			${SAND_LDFLAGS}
 			${SIGC_LDFLAGS}
+			${PMLOG_LDFLAGS}
 			${Boost_LIBRARIES}
 			${GLIB_2_LDFLAGS})
 
